### PR TITLE
Refactor check logic of index parameters

### PR DESCRIPTION
**Pull request by longjiquan**
issue: #23497

### CODE REVIEW

**Comment by longjiquan** [comment]
> Check for unnecessary parameters?

Not sure if this check is required.

File: internal/proxy/task_index.go [hunk 1]
```diff
@@ -242,26 +242,22 @@ func checkTrain(field *schemapb.FieldSchema, indexParams map[string]string) erro
...
 	if err := fillDimension(field, indexParams); err != nil {
 		return err
 	}
 
-	ok := adapter.CheckValidDataType(field.GetDataType())
-	if !ok {
-		log.Warn("Field data type don't support the index build type", zap.String("fieldDataType", field.GetDataType().String()), zap.String("indexType", indexType))
-		return fmt.Errorf("field data type %s don't support the index build type %s", field.GetDataType().String(), indexType)
+	if err := checker.CheckValidDataType(field.GetDataType()); err != nil {
+		return err
```
Comment (czs007) [inline internal/proxy/task_index.go:256]: add Info/Warn log?

File: pkg/util/indexparamcheck/bin_flat_checker_test.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	}
+	paramsWithoutDim := map[string]string{
+		Metric: JACCARD,
+	}
+	cases := []struct {
+		params   map[string]string
+		errIsNil bool
+	}{
+		{validParams, true},
+		{paramsWithoutDim, false},
```
Comment (czs007) [inline pkg/util/indexparamcheck/bin_flat_checker_test.go:23]: add test cases for invalid metric, DataType, etc ?

File: pkg/util/indexparamcheck/bin_ivf_flat_checker_test.go [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+	cases := []struct {
+		params   map[string]string
+		errIsNil bool
+	}{
+		{validParams, true},
+		{paramsWithoutDim, false},
+		{invalidIVFParamsMin(), false},
+		{invalidIVFParamsMax(), false},
+		{invalidParams, false},
```
Comment (czs007) [inline pkg/util/indexparamcheck/bin_ivf_flat_checker_test.go:36]: Ditto

File: pkg/util/indexparamcheck/diskann_checker_test.go [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+	invalidParamsSmallDim[DIM] = strconv.Itoa(15)
+
+	cases := []struct {
+		params   map[string]string
+		errIsNil bool
+	}{
+		{validParams, true},
+		{validParamsBigDim, true},
+		{invalidParamsWithoutDim, false},
+		{invalidParamsSmallDim, false},
```
Comment (czs007) [inline pkg/util/indexparamcheck/diskann_checker_test.go:32]: add invalid metric_type test cases

File: pkg/util/indexparamcheck/hnsw_checker_test.go [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+	invalidEfParamsMin[EFConstruction] = strconv.Itoa(HNSWMinEfConstruction - 1)
+
+	invalidEfParamsMax := copyParams(validParams)
+	invalidEfParamsMax[EFConstruction] = strconv.Itoa(HNSWMaxEfConstruction + 1)
+
+	invalidMParamsMin := copyParams(validParams)
+	invalidMParamsMin[HNSWM] = strconv.Itoa(HNSWMinM - 1)
+
+	invalidMParamsMax := copyParams(validParams)
+	invalidMParamsMax[HNSWM] = strconv.Itoa(HNSWMaxM + 1)
```
Comment (czs007) [inline pkg/util/indexparamcheck/hnsw_checker_test.go:29]: invalid metric type

File: pkg/util/indexparamcheck/ivf_base_checker_test.go [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+		Metric: L2,
+	}
+
+	cases := []struct {
+		params   map[string]string
+		errIsNil bool
+	}{
+		{validParams, true},
+		{invalidIVFParamsMin(), false},
+		{invalidIVFParamsMax(), false},
```
Comment (czs007) [inline pkg/util/indexparamcheck/ivf_base_checker_test.go:23]: invalid metric_type

File: pkg/util/indexparamcheck/ivf_pq_checker_test.go [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+		{validParamsWithoutNbits, true},
+		{invalidIVFParamsMin(), false},
+		{invalidIVFParamsMax(), false},
+		{validParamsWithoutDim, false},
+		{invalidParamsDim, false},
+		{invalidParamsNbits, false},
+		{invalidParamsWithoutIVF, false},
+		{invalidParamsIVF, false},
+		{invalidParamsM, false},
+		{invalidParamsMzero, false},
```
Comment (czs007) [inline pkg/util/indexparamcheck/ivf_pq_checker_test.go:69]: add invalid metric_type test cases.

File: pkg/util/indexparamcheck/ivf_pq_checker.go [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+	}
+	m, err := strconv.Atoi(mStr)
+	if err != nil || m == 0 { // invalid m
+		return fmt.Errorf("invalid `m`: %s", mStr)
+	}
+
+	return c.checkCPUPQParams(dimension, m)
+}
+
+func (c *ivfPQChecker) checkGPUPQParams(dimension, m, nbits int) error {
```
Comment (czs007) [inline pkg/util/indexparamcheck/ivf_pq_checker.go:56]: remove this unused function.

**Pull request merged**